src/datahike/connector.cljc

```diff
@@ -0,0 +1,129 @@
+        :schema schema
+        :max-eid (db/init-max-eid eavt-key)
+        :eavt eavt-key
+        :aevt aevt-key
+        :avet avet-key
+        :rschema rschema
+        :store store
+        :uri uri))))
+
+(defn release [conn]
+  (let [[m store-scheme path] (parse-uri (:uri @conn))]
+    (ds/release-store store-scheme (:store @conn))))
+
+(defn transact [connection tx-data]
+  {:pre [(d/conn? connection)]}
+  (future
+    (locking connection
+      (let [{:keys [db-after] :as tx-report} @(d/transact connection tx-data)
+            {:keys [eavt aevt avet schema rschema config]} db-after
+            store (:store @connection)
+            backend (kons/->KonserveBackend store)
+            eavt-flushed (di/-flush eavt backend)
+            aevt-flushed (di/-flush aevt backend)
+            avet-flushed (di/-flush avet backend)]
+        (<?? S (k/assoc-in store [:db]
+                           {:schema   schema
+                            :rschema  rschema
+                            :config   config
+                            :eavt-key eavt-flushed
+                            :aevt-key aevt-flushed
+                            :avet-key avet-flushed}))
+        (reset! connection (assoc db-after
+                             :eavt eavt-flushed
+                             :aevt aevt-flushed
+                             :avet avet-flushed))
+        tx-report))))
+
+(defn transact! [connection tx-data]
+  (try
+    (deref (transact connection tx-data))
+    (catch Exception e
+      (throw (.getCause e)))))
+
+(defmulti create-database
+          "Creates a new database"
+          {:arglists '([config])}
+          (fn [config] (type config)))
+
+(defmethod create-database String [uri]
+  (create-database {:uri uri}))
+
+(defmethod create-database clojure.lang.PersistentArrayMap
+  [{:keys [uri initial-tx schema-on-read]}]
+  (let [[m store-scheme path] (parse-uri uri)
+        _ (when-not m
+            (throw (ex-info "URI cannot be parsed." {:uri uri})))
+         store (kc/ensure-cache
+                  (ds/empty-store store-scheme path)
+                  (atom (cache/lru-cache-factory {} :threshold 1000)))
+        stored-db (<?? S (k/get-in store [:db]))
+        _ (when stored-db
+            (throw (ex-info "Database already exists." {:type :db-already-exists :uri uri})))
+        config {:schema-on-read (or schema-on-read false)}
+        {:keys [eavt aevt avet schema rschema config]} (db/empty-db {:db/ident {:db/unique :db.unique/identity}} (ds/scheme->index store-scheme) :config config)
+        backend (kons/->KonserveBackend store)]
+    (<?? S (k/assoc-in store [:db]
+                       {:schema   schema
+                        :rschema  rschema
+                        :config   config
+                        :eavt-key (di/-flush eavt backend)
+                        :aevt-key (di/-flush aevt backend)
+                        :avet-key (di/-flush avet backend)}))
+    (ds/release-store store-scheme store)
```

Comment:
I find it a bit surprising that just after creating the store, we have to already release it. May be it is just a naming issue.